src/lava/lib/optimization/utils/generators/mis.py

```diff
@@ -0,0 +1,180 @@
+        adj_mc = np.logical_not(adjacency_matrix)
+        adj_mc = adj_mc.astype(int)
+        adj_mc = adj_mc - np.diag(adj_mc.diagonal())
+        return adj_mc
+
+    @staticmethod
+    def _get_qubo_cost_from_adjacency(adjacency_matrix: np.ndarray,
+                                      w_diag: float,
+                                      w_off: float) -> np.ndarray:
+        if w_off <= 2 * w_diag:
+            raise ValueError(
+                "Off-diagonal weights must be > 2 x diagonal weights.")
+
+        # Translate the connectivity matrix to a QUBO matrix
+        num_variables = adjacency_matrix.shape[0]
+        q = - w_diag * np.eye(num_variables) + w_off / 2 * adjacency_matrix
+        return q.astype(int)
+
+    def get_graph(self) -> netx.Graph:
+        """Returns the graph in networkx format."""
+        graph = self._get_graph_from_adjacency_matrix(self._adjacency)
+        return graph
+
+    def get_graph_matrix(self) -> np.ndarray:
+        """Returns the adjacency matrix of the graph."""
+        return self._adjacency
+
+    def get_complement_graph(self) -> netx.Graph:
+        """Returns the complement graph in networkx format."""
+        c_adjacency = self._get_adjacency_of_complement_graph(self._adjacency)
+        c_graph = self._get_graph_from_adjacency_matrix(c_adjacency)
+        return c_graph
+
+    def get_complement_graph_matrix(self) -> np.ndarray:
+        """Returns the adjacency matrix of the complement graph."""
+        c_adjacency = self._get_adjacency_of_complement_graph(self._adjacency)
+        return c_adjacency
+
+    def get_as_qubo(self, w_diag: float, w_off: float) -> QUBO:
+        """
+        Creates a QUBO whose solution corresponds to the maximum independent
+        set (MIS) of the graph defined by the input adjacency matrix.
+
+        The goal of the QUBO is to minimize the cost
+            min x^T * Q * x ,
+        where the vector x is defined as:
+            x_i = 1 if vertex i is part of the MIs
+            x_i = 0 if vertex i is not part of the MIS,
+        and the QUBO matrix is given by
+            Q_ii = w_diag
+            Q_ij = w_off (for i~=j) .
+        """
+        q = self._get_qubo_cost_from_adjacency(self._adjacency, w_diag, w_off)
+        qubo = QUBO(q)
+        return qubo
+
+    def _get_random_graph_mtx(self) -> np.ndarray:
+        """
+        Creates an undirected graph with random connectivity between nodes
+        and returns its adjacency matrix.
+        """
+
+        np.random.seed(self.seed)
+
+        random_matrix = np.random.rand(self.num_vertices, self.num_vertices)
+        adjacency = (random_matrix < self.connection_prob).astype(int)
+
+        adjacency = np.triu(adjacency)
+        # ensure that the matrix is symmetric
+        adjacency += adjacency.T - 2 * np.diag(adjacency.diagonal())
+
+        return adjacency
+
+
+def find_maximum_independent_set(mis_problem: MISProblem) -> \

```

Comment:
Maybe better to make this a method of the MIS class?

Reply:
Sure!